File: src/main/java/org/gtreimagined/gtlib/gui/slot/SlotInput.java
package org.gtreimagined.gtlib.gui.slot;

import org.gtreimagined.gtlib.capability.IGuiHandler;
import org.gtreimagined.gtlib.gui.SlotType;
import net.minecraft.world.entity.player.Player;
import net.minecraftforge.items.IItemHandler;

public class SlotInput extends AbstractSlot<SlotInput> {

    public SlotInput(SlotType<SlotInput> type, IGuiHandler tile, IItemHandler stackHandler, int index) {
        super(type, tile, stackHandler, index);
    }

    @Override
    public boolean mayPickup(Player player) {
        return true;
    }
}

File: src/main/java/org/gtreimagined/gtlib/gui/slot/SlotInput.kt
package org.gtreimagined.gtlib.gui.slot

import net.minecraft.world.entity.player.Player
import net.minecraftforge.items.IItemHandler
import org.gtreimagined.gtlib.capability.IGuiHandler
import org.gtreimagined.gtlib.gui.SlotType

class SlotInput(type: SlotType<SlotInput>, tile: IGuiHandler, stackHandler: IItemHandler, index: Int) :
    AbstractSlot<SlotInput>(type, tile, stackHandler, index) {
    override fun mayPickup(player: Player): Boolean {
        return true
    }
}
